plugins/storage/rdbms/report.go

```diff
@@ -163,3 +165,25 @@ func (r *RDBMS) GetJobReport(jobID types.JobID) (*job.JobReport, error) {
 		FinalReports: finalReports,
 	}, nil
 }
+
+// StoreTestSteps persists a list of test step descriptors to the database.
+func (r *RDBMS) StoreTestSteps(jobID types.JobID, steps []*test.TestStepDescriptor) error {
```

Comment:
`RDBMS` now supports transactions, and will need to lock all the `Exec`, `Query` and row scanning operations, in case there is an active transaction, so there should be:

```
r.lockTx()
defer r.unlockTx()
```

at the beginning of `StoreTestSteps`. If it wasn't for having to protect rows scanning, this could be hidden behind an interface which wraps `Query` and `Exec`, but unfortunately that's not enough, everything needs to be locked and serialized.

The same applies to `GetTestSteps` below.

Reply:
After going through the whole patch, I am not sure I understand why we need `StoreTestSteps`? When we invoke `StoreJobRequest` we already have the `[][]*test.TestStepDescriptor` (for all tests), so we should be able to persist everything we need in the `job` table via `StoreJobRequest`?

Reply:
I've changed approach a couple of times and `StoreJobRequest` is probably a leftover. Let me fix the merge conflicts first, then I'll go through the diff again to remove what should not be there. Sorry!